shared/Microsoft.AspNetCore.HttpSys.Sources/NativeInterop/HttpApiTypes.cs

```diff
@@ -0,0 +1,694 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+namespace Microsoft.AspNetCore.HttpSys.Internal
+{
+    internal static unsafe class HttpApiTypes
+    {
+        internal enum HTTP_API_VERSION
+        {
+            Invalid,
+            Version10,
+            Version20,
+        }
+
+        // see http.w for definitions
+        internal enum HTTP_SERVER_PROPERTY
+        {
+            HttpServerAuthenticationProperty,
+            HttpServerLoggingProperty,
+            HttpServerQosProperty,
+            HttpServerTimeoutsProperty,
+            HttpServerQueueLengthProperty,
+            HttpServerStateProperty,
+            HttpServer503VerbosityProperty,
+            HttpServerBindingProperty,
+            HttpServerExtendedAuthenticationProperty,
+            HttpServerListenEndpointProperty,
+            HttpServerChannelBindProperty,
+            HttpServerProtectionLevelProperty,
+        }
+
+        // Currently only one request info type is supported but the enum is for future extensibility.
+
+        internal enum HTTP_REQUEST_INFO_TYPE
+        {
+            HttpRequestInfoTypeAuth,
+            HttpRequestInfoTypeChannelBind,
+            HttpRequestInfoTypeSslProtocol,
+            HttpRequestInfoTypeSslTokenBinding
+        }
+
+        internal enum HTTP_RESPONSE_INFO_TYPE
+        {
+            HttpResponseInfoTypeMultipleKnownHeaders,
+            HttpResponseInfoTypeAuthenticationProperty,
+            HttpResponseInfoTypeQosProperty,
+        }
+
+        internal enum HTTP_TIMEOUT_TYPE
+        {
+            EntityBody,
+            DrainEntityBody,
+            RequestQueue,
+            IdleConnection,
+            HeaderWait,
+            MinSendRate,
+        }
+
+        internal const int MaxTimeout = 6;
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct HTTP_VERSION
+        {
+            internal ushort MajorVersion;
+            internal ushort MinorVersion;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct HTTP_KNOWN_HEADER
+        {
+            internal ushort RawValueLength;
+            internal sbyte* pRawValue;
```

Comment:
Can we change this to byte*?

Reply:
I don't know why it was sbyte to begin with. Try byte and see if the tests notice.